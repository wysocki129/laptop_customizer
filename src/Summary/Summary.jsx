import React, { Component } from 'react';
import './Summary.css';

class Summary extends Component {
	render() {
		const { selectedParts } = this.props;
		const laptopSummary = Object.keys(selectedParts).map(key => (
			<div className="summary__option" key={key}>
				<div className="summary__option__label">{key} </div>
				<div className="summary__option__value">
					{selectedParts[key].name}
				</div>
				<div className="summary__option__cost">
					{new Intl.NumberFormat('en-US', {
						style: 'currency',
						currency: 'USD'
					}).format(selectedParts[key].cost)}
				</div>
			</div>
		));
		const total = Object.keys(selectedParts).reduce(
			(acc, curr) => acc + selectedParts[curr].cost,
			0
		);
		return (
			<section className="main__summary">
				<h3>NEW GREENLEAF 2018</h3>
				{laptopSummary}
				<div className="summary__total">
					<div className="summary__total__label">
						Your Price:{' '}
					</div>
					<div className="summary__total__value">
						{new Intl.NumberFormat('en-US', {
							style: 'currency',
							currency: 'USD'
						}).format(total)}
					</div>
				</div>
			</section>
		);
	}
}
Summary.defaultProps = {
	selectedParts: []
};

export default Summary;
