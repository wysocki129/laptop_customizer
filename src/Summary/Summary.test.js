import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Summary from './Summary';

const SELECTEDTEST = {
	selected: {
		Processor: {
			name:
				'17th Generation Intel Core HB (7 Core with donut spare)',
			cost: 700
		},
		'Operating System': {
			name: 'Ubuntu Linux 16.04',
			cost: 200
		},
		'Video Card': {
			name: 'Toyota Corolla 1.5v',
			cost: 1150.98
		},
		Display: {
			name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
			cost: 1500
		}
	}
};
describe('Summary Component', () => {
	it('renders without crashing the app', () => {
		shallow(<Summary />);
	});
	it('displays the correct total', () => {
		const wrapper = shallow(<Summary selectedParts={SELECTEDTEST} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
