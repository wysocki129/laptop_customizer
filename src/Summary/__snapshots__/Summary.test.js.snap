// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Component displays the correct total 1`] = `
<section
  className="main__summary"
>
  <h3>
    NEW GREENLEAF 2018
  </h3>
  <div
    className="summary__option"
    key="selected"
  >
    <div
      className="summary__option__label"
    >
      selected
       
    </div>
    <div
      className="summary__option__value"
    />
    <div
      className="summary__option__cost"
    >
      $NaN
    </div>
  </div>
  <div
    className="summary__total"
  >
    <div
      className="summary__total__label"
    >
      Your Price:
       
    </div>
    <div
      className="summary__total__value"
    >
      $NaN
    </div>
  </div>
</section>
`;
